lib/internal/http2/core.js

```diff
@@ -1671,16 +1652,12 @@ class Http2Stream extends Duplex {
     if (!this.headersSent)
       this[kProceed]();
 
-    const handle = this[kHandle];
-    const req = new WriteWrap();
+    const req = createWriteWrap(this[kHandle], afterDoStreamWrite);
     req.stream = this[kID];
-    req.handle = handle;
     req.callback = cb;
-    req.oncomplete = afterDoStreamWrite;
-    req.async = false;
-    const err = createWriteReq(req, handle, data, encoding);
-    if (err)
-      return this.destroy(errnoException(err, 'write', req.error), cb);
+
+    writeGeneric(this, req, data, encoding, cb);
```

Comment:
Would be nice to have this done for `writev` too :)

Reply:
@addaleax done 👍